mllib/src/main/scala/org/apache/spark/ml/linalg/JsonMatrixConverter.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml.linalg
+
+import org.json4s.DefaultFormats
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods.{compact, parse => parseJson, render}
+
+private[ml] object JsonMatrixConverter {
+
+  /** Unique class name for identifying JSON object encoded by this class. */
+  val className = "org.apache.spark.ml.linalg.Matrix"
```

Comment:
I added this as an identifier, so during loading we know which JSON converter to invoke. Yet I didn't add it for JsonVectorConverter, to avoid breaking old models.

Reply:
Or can we just use ```type``` to identify vector and matrix? For example, ```type``` less than 10 is reserved for vector and more than 10 is for matrix. What do you think of it?

Reply:
Sure. I can see your point. An example for the json,
{"class":"org.apache.spark.ml.linalg.Matrix","type":0,"numRows":3,"numCols":2,"colPtrs":[0,1,3,4],"rowIndices":[1,0,1,0],"values":[1.21,2.3,9.8,9.0],"isTransposed":true}

Sure we can use Int for `class` or different ranges of `type` to distinguish different classes. While storage efficiency is important, I also want to have a solution that's easy to extend for future and friendly for debugging or trouble shooting. Thus keeping the literal class info in the json is still attractive to me. Is it OK if we simply use `class:Matrix` for this?

Reply:
I think `class: "matrix"` is OK. shorter identifier is better.